migrations/20170115_api_date_validation.js

```diff
@@ -0,0 +1,65 @@
+var migrationName = '20170115_api_date_validation.js';
+var authorName = 'sprusr'; // in case script author needs to know when their ...
+var authorUuid = 'db711097-c5d0-4d25-8e5e-d955da4b79fd'; //... own data is done
+
+/*
+ * API date validation
+ */
+
+var monk = require('monk');
+
+var connectionString = 'mongodb://localhost:27017/habitrpg?auto_reconnect=true'; // FOR TEST DATABASE
+
+var dbTasks = monk(connectionString).get('tasks', { castIds: false });
+
+var query = {
+  date: { $exists: true }
+};
+
+console.warn('Updating tasks...');
+
+dbTasks.find(query, 'date').each(function(task, cursor) {
+  var updateQuery = {};
+  var newDate = new Date(task.date);
+
+  if(isValidDate(newDate)) {
+    updateQuery = {
+      $set: {
+        'date': newDate
+      }
+    }
+  } else {
+    updateQuery = {
+      $unset: {
+        'date': ''
+      }
+    }
+  }
+
+  console.log(task, updateQuery);
+
+  cursor.pause();
```

Comment:
what is this for? To make sure only one task is processed at a certain time? There are more than 20 milions tasks that have this field so even with a low processing time of 5ms per task (and with latency it's going to be a lot more) it'll take more than a day to run the migration. I think we want multiple tasks processed at the same time. For example fetch 100 tasks, process them while we fetch the next 10 and when the first 10 have been processed repeat the process